libobs/util/platform-windows.c

```diff
@@ -1088,9 +1088,24 @@ bool is_64_bit_windows(void)
 #endif
 }
 
+bool is_arm64_windows(void)
+{
+#if defined(_M_ARM64) || defined(_M_ARM64EC)
+	return true;
+#else
+	USHORT _, nativeMachine;
+	auto result = IsWow64Process2(GetCurrentProcess(), &_, &nativeMachine);
```

Comment:
The new C auto is not implemented until C23, which we do not currently support. This is currently the old C auto, which we do not recommend. Use the actual type here.